packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -286,17 +285,37 @@ const EDiscovery = WebexPlugin.extend({
     options = {...defaultOptions, ...options};
 
     // If the content summary for this report has already been cached then it can be retrieved from there instead of making a network call to ediscovery
-    if (this.contentSummaryCache.has(reportId) && this.contentSummaryCache.get(reportId).has(spaceId)) {
-      return this.contentSummaryCache.get(reportId).get(spaceId);
+    if (this.contentSummaryBySpaceIdCache.has(reportId) && this.contentSummaryBySpaceIdCache.get(reportId).has(spaceId)) {
+      return this.contentSummaryBySpaceIdCache.get(reportId).get(spaceId);
     }
 
+    this.logger.warn(`Cache miss for space ${spaceId} in getContentSummaryBySpaceId`);
+
     return this.request({
       method: 'GET',
       service: 'ediscovery',
       resource: `reports/${reportId}/contents/summary/${spaceId}`,
       timeout: options.timeoutMs
     })
-      .then((res) => res.body);
+      .then((res) => {
+        if (!this.contentSummaryBySpaceIdCache.has(reportId)) {
+          this.contentSummaryBySpaceIdCache.set(reportId, new Map());
+        }
+
+        if (res.body && res.body.spaceId) {
+          try {
+            this.contentSummaryBySpaceIdCache.get(reportId).set(res.body.spaceId, res.body);
+          }
+          catch (error) {
+            this.logger.error(`Error adding ${res.body.spaceId} to contentSummaryBySpaceIdCache: ${error}`);
+          }
+        }
+        else {
+          this.logger.error('Error adding undefined space to contentSummaryBySpaceIdCache');
+        }
+
+        return res.body;
+      });
```

Comment:
Good idea to handle the `catch` scenario? Or is it more appropriate to let the upper level handle it?

Reply:
I think we should let the upper layer handle it, that seems to be the general pattern adopted by the other plugins. If there are special cases we need to handle (e.g. the invalid emails in createReport), we can add catch handlers for them, but otherwise it's best to keep this class as straightforward as possible.